plugins/autoid/plugin.js

```diff
@@ -1,9 +1,95 @@
 'use strict';
 
 (function () {
-	CKEDITOR.plugins.add('autoid', {
-		init: function (editor) {
+  var editorHasFocus = false;
+  var commandIsActive = false;
+  var EVENT_NAMES = {
+    ALL_IDS_COMPLETE: 'allIdsComplete',
+    ID_ADDED: 'idAdded'
+  };
 
-		}
-	});
+  CKEDITOR.plugins.add('autoid', {
+    init: function (editor) {
+      var self = this;
+
+      editor.addFeature({
+        allowedContent: 'h1 h2 h3 h4 h5 h6 *[id]'
+      });
+
+      if (editor && !editor.config.autoid) {
+        editor.config.autoid = {};
+      }
+      self.settings = editor.config.autoid;
+      if (!self.settings) {
+        self.settings = {};
+      }
+
+      editor.addCommand('autoid', {
+        exec: function (editor) {
+          if (!commandIsActive) {
+            start();
+          } else {
+            stop();
+          }
+        },
+        editorFocus: true
+      });
+
+      editor.on("instanceReady", function () {
+        if (self.settings.autostart !== false) {
```

Comment:
post-standup, the confusion is now resolved:

> [8:42 AM] Seth Baughman: sweet
[8:43 AM] Seth Baughman: so question... do you not want the autostart to be the default?
[8:44 AM] Alex Tsai: Actually, you are right, it probably should be the default.  I see where the confusion happened
[8:44 AM] Alex Tsai: we both fixed the tests,
[8:44 AM] Alex Tsai: I fixed them by not autostarting by changing the default
[8:44 AM] Alex Tsai: you just changed the config
[8:44 AM] Alex Tsai: yours is probably more like the expected behaviour